sycl/source/detail/device_impl.hpp

```diff
@@ -59,6 +59,17 @@ template <typename> static constexpr bool is_std_vector_v = false;
 template <typename T>
 static constexpr bool is_std_vector_v<std::vector<T>> = true;
 
+template <ur_device_info_t Desc> static constexpr auto ur_ret_type_impl() {
+  if constexpr (false) {
+  }
```

Comment:
Nit; Do we actually need this? If all cases using `MAP` return, then we shouldn't need the `else`, right?

Reply:
Yes, because `MAP` starts with `else if`. `#else == #if - 1`